File: xcode-core/examples/python/class_methods.py
"""类方法 trace 示例：trace 类的方法 + 调用方位置"""


class Calculator:
    """A simple calculator class."""

    def add(self, a: int, b: int) -> int:
        return a + b

    def multiply(self, a: int, b: int) -> int:
        return a * b

    def chain(self, x: int) -> int:
        # 调用方位置：第 25 行
        result = self.add(x, 1)
        # 调用方位置：第 27 行
        result = self.multiply(result, 2)
        return result


if __name__ == '__main__':
    calc = Calculator()
    # 调用方位置：第 34 行
    result = calc.chain(5)
    print(f"Result: {result}")

File: xcode-core/examples/rust/src/class_methods.rs
// 结构体方法 trace 示例

struct Calculator {
    name: String,
}

impl Calculator {
    fn add(&self, a: i32, b: i32) -> i32 {
        a + b
    }

    fn multiply(&self, a: i32, b: i32) -> i32 {
        a * b
    }

    fn chain(&self, x: i32) -> i32 {
        let r1 = self.add(x, 1);
        let r2 = self.multiply(r1, 2);
        r2
    }
}

fn main() {
    let calc = Calculator {
        name: "test".to_string(),
    };
    let result = calc.chain(5);
    println!("Result: {}", result);
}
